pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -299,3 +302,26 @@ type ConsulSDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// DNSSDConfig allows specifying a set of DNS domain names which are periodically queried to discover a list of targets.
+// The DNS servers to be contacted are read from /etc/resolv.conf.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#dns_sd_config
+// +k8s:openapi-gen=true
+type DNSSDConfig struct {
+	// A list of DNS domain names to be queried.
+	// +kubebuilder:validation:MinItems:=1
+	Names []string `json:"names"`
+	// RefreshInterval configures the time after which the provided names are refreshed.
+	// If unset, Prometheus uses its default value (30s).
```

Comment:
(nit) let's keep the same wording used elsewhere and avoid mentioning the exact default value.
```suggestion
	// If not set, Prometheus uses its default value.
```